5167: Added ISAPI to local k8 cluster

adrian-fitsios opened the pull request:
- added isapi, isapi-mssql, isapi-sample to local cluster
- renamed mssql server (used by bapi) to be less generic
- added subfolder for standing up cluster without isapi
- added subfolder for standing up cluster without isapi on a mac
- amended readme to include setting up client secret which is needed for isapi
- reduced cpu allocations for containers from 500m to 250m as lower-end machines might not be able to stand everything up otherwise

adrian-fitsios committed:
5167: Added ISAPI to local k8 cluster

- added isapi, isapi-mssql, isapi-sample to local cluster
- renamed mssql server (used by bapi) to be less generic
- added subfolder for standing up cluster without isapi
- added subfolder for standing up cluster without isapi on a mac
- amended readme to include setting up client secret which is needed for isapi
- reduced cpu allocations for containers from 500m to 250m as lower-end machines might not be able to stand everything up otherwise

stuart-bass-cgi commented on local-kubernetes/README.md:
```diff
@@ -10,7 +10,13 @@ Create the buying catalogue namespace - `kubectl apply -f local-namespace.yml`
+- `kubectl create secret generic mssql --from-literal=SA_PASSWORD="<YOUR PASSWORD>" -n buyingcatalogue`. **Password must be follow this [policy](https://docs.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=sql-server-ver15#password-complexity)**.
+
+### identity server:
+- `kubectl create secret generic oidc-client-secret --from-literal=OIDC_CLIENT_SECRET="<YOUR CLIENT SECRET>" -n buyingcatalogue`
```
does this client secret need to be something in particular to work with the front end?

adrian-fitsios replied:
The front-end and back-end are configured to both use the same secret, I didn't assume it needs to be something specific as that would defeat the purpose of a secret. I went ahead and tested changing this secret to some arbitrary string and - to my surprise it did not work, however it did not work because this env. variable injected by k8 does not override the 'clients': { ..., 'secret': 'value' } in Isapi's appconfig.json which looks like an issue with Isapi. When Isapi is taken outside of the cluster and the value in appsettings is set to match the kube secret everything is fine. I'll try to get Isapi to behave accordingly (change the secret's value to come from env. var if present) and that will fix this.

adrian-fitsios replied:
Update: 
Turns out I glazed over the fact that isapi's appsettings has an **array** of clients, therefore the env variable set in the yaml manifest was't overriding anything.

stuart-bass-cgi commented on local-kubernetes/isapi-mac/local-isapi-kubernetes-manifest.yml:
```diff
@@ -0,0 +1,65 @@
+        - name: CONNECTIONSTRINGS__CATALOGUEUSERS
+          value: Data Source=isapi-mssql-deployment.buyingcatalogue,1451;Initial Catalog=CatalogueUsers;MultipleActiveResultSets=True;User Id=NHSD;Password=DisruptTheMarket1!
+        - name: ISSUERURL
+          value: http://host.docker.internal:5102
```
shouldn't this be http://docker.for.mac.localhost/?

adrian-fitsios replied:
No need to change this file as its taken out of launch-env.. scripts.

stuart-bass-cgi commented on local-kubernetes/isapi/local-isapi-kubernetes-manifest.yml:
```diff
@@ -0,0 +1,65 @@
+        - name: CONNECTIONSTRINGS__CATALOGUEUSERS
+          value: Data Source=isapi-mssql-deployment.buyingcatalogue,1451;Initial Catalog=CatalogueUsers;MultipleActiveResultSets=True;User Id=NHSD;Password=DisruptTheMarket1!
+        - name: ISSUERURL
+          value: http://host.docker.internal:5102
```
http://docker.for.win.localhost/?

adrian-fitsios replied:
No need to change this file as its taken out of launch-env.. scripts.

stuart-bass-cgi commented on local-kubernetes/isapi/local-isapi-sample-kubernetes-manifest.yml:
```diff
@@ -0,0 +1,166 @@
+apiVersion: apps/v1
```
this file looks like a duplicate?

adrian-fitsios replied:
Its not, have a closer look. This is the sample isapi - includes (isapi, sample resource and sample mvc)

stuart-bass-cgi commented on local-kubernetes/launch-environment.ps1:
```diff
@@ -1,7 +1,10 @@
 kubectl apply -f local-dapi-kubernetes-manifest.yml
+kubectl apply -f local-isapi-sql-kubernetes-manifest.yml
+kubectl apply -f local-isapi-kubernetes-manifest.yml
+#kubectl apply -f local-isapi-sample-kubernetes-manifest.yml
```
why commented out?

adrian-fitsios replied:
Runs on same port as the actual isapi. I thought we want to keep this sample isapi service as a reference to how things can be wired up and not an actual service to stand up with everything else.

stuart-bass-cgi commented on local-kubernetes/launch-environment.sh:
```diff
@@ -2,9 +2,12 @@
 kubectl apply -f local-dapi-kubernetes-manifest.yml
+kubectl apply -f local-isapi-sql-kubernetes-manifest.yml
+kubectl apply -f local-isapi-kubernetes-manifest.yml
+#kubectl apply -f local-isapi-sample-kubernetes-manifest.yml
```
why commented out?

adrian-fitsios replied:
Runs on same port as the actual isapi. I thought we want to keep this sample isapi service as a reference to how things can be wired up and not an actual service to stand up with everything else.

stuart-bass-cgi commented on local-kubernetes/local-isapi-kubernetes-manifest.yml:
```diff
@@ -0,0 +1,65 @@
+        - name: CONNECTIONSTRINGS__CATALOGUEUSERS
+          value: Data Source=isapi-mssql-deployment.buyingcatalogue,1451;Initial Catalog=CatalogueUsers;MultipleActiveResultSets=True;User Id=NHSD;Password=DisruptTheMarket1!
+        - name: ISSUERURL
+          value: http://host.docker.internal:5102
```
http://docker.for.win.localhost//?

adrian-fitsios replied:
No need to change this file as its taken out of launch-env.. scripts.

stuart-bass-cgi commented on local-kubernetes/local-isapi-sample-kubernetes-manifest.yml:
```diff
@@ -0,0 +1,166 @@
+apiVersion: apps/v1
```
same as above comment - why include this?

adrian-fitsios replied:
as above replies.

stuart-bass-cgi commented on local-kubernetes/local-isapi-sql-kubernetes-manifest.yml:
```diff
@@ -0,0 +1,46 @@
+        - name: DB_NAME
+          value: "CatalogueUsers"
+        - name: NHSD_PASSWORD
+          value: "DisruptTheMarket1!"
```
eof new line?

adrian-fitsios committed:
5167: PR Comments

- actually inject the clients secret from kube's secrets
- new lines at eof
- removed helper script
- readme update

adrian-fitsios committed:
5167: PR comments

- removed sample isapi
- updated pb subfolder to work with pb running locally (outside cluster)
